dulwich/tests/test_porcelain.py

```diff
@@ -327,6 +328,38 @@ def test_add_not_in_repo(self):
             paths=["../foo"])
         self.assertEqual([], list(self.repo.open_index()))
 
+    def test_add_all_base(self):
+        os.mkdir(os.path.join(self.repo.path, 'foo'))
+        with open(os.path.join(self.repo.path, 'blah'), 'w') as f:
+            f.write("\n")
+        with open(os.path.join(self.repo.path, 'foo', 'blie'), 'w') as f:
+            f.write("\n")
+
+        cwd = os.getcwd()
+        try:
+            os.chdir(os.path.join(self.repo.path, 'foo'))
+            porcelain.add(repo=self.repo.path, all=True)
+            porcelain.commit(repo=self.repo.path, message=b'test',
+                             author=b'test <email>',
+                             committer=b'test <email>')
+        finally:
+            os.chdir(cwd)
+
+    def test_add_all_and_paths(self):
+        os.mkdir(os.path.join(self.repo.path, 'foo'))
+        with open(os.path.join(self.repo.path, 'blah'), 'w') as f:
+            f.write("\n")
+        with open(os.path.join(self.repo.path, 'foo', 'blie'), 'w') as f:
+            f.write("\n")
+
+        porcelain.add(repo=self.repo.path, all=True,
```

Comment:
Shouldn't this verify that the changes to a file are also added if one of the files existed before?